lib/Page.js

```diff
@@ -269,6 +270,26 @@ class Page extends EventEmitter {
     return context.evaluateHandle(pageFunction, ...args);
   }
 
+  /**
+   * @param {string} contextIdentifier
+   * @returns {!Promise.<!Puppeteer.ExecutionContext>}
+   */
+  async createNewIsolatedContext(contextIdentifier) {
```

Comment:
Does `contextIdentifier` have any significance for puppeteer scripts? It looks like this parameter is unnecessary.

Reply:
@JoelEinbinder: it'll play nice with your "frame.executionContexts()" pull request: #2812 

@furstenheim: nit - I'd name this method `createIsolatedWorld(name)`.

Reply:
The name should at least be optional in that case.

Reply:
@JoelEinbinder should name be created as a random string then?

Reply:
@furstenheim I think name is not necessary for the protocol method.